thewalrus/quantum.py

```diff
@@ -1086,12 +1088,24 @@ def probabilities(mu, cov, cutoff, hbar=2.0, rtol=1e-05, atol=1e-08):
     if is_pure_cov(cov, hbar=hbar, rtol=rtol, atol=atol):  # Check if the covariance matrix cov is pure
         return np.abs(state_vector(mu, cov, cutoff=cutoff, hbar=hbar, check_purity=False)) ** 2
     num_modes = len(mu) // 2
-    probs = np.zeros([cutoff] * num_modes)
-    for i in product(range(cutoff), repeat=num_modes):
-        probs[i] = np.maximum(
-            0.0, np.real_if_close(density_matrix_element(mu, cov, i, i, hbar=hbar))
-        )
+
+    if parallel:
+        compute_list = []
+        # Create a list of parallelizable computations
+        for i in product(range(cutoff), repeat=num_modes):
+            compute_list.append(dask.delayed(density_matrix_element)(mu, cov, i, i, hbar=hbar))
+
+        probs = np.maximum(
+            0.0, np.real_if_close(dask.compute(*compute_list, scheduler="threads"))
+        ).reshape([cutoff] * num_modes)

```

Comment:
Would be curious if this is faster in practice! partly because `density_matrix_element` is already parallelized, right? iirc it uses OpenMP to compute the hafnian corresponding to a single matrix element, so there is almost a double-parallelization happening

Reply:
This might be correct. I noticed that it didn't seem to provide any speedups (but only after I pushed it to this PR), but I haven't really benchmarked it yet.